src/v/cluster/partition.cc

```diff
@@ -451,6 +457,12 @@ ss::future<> partition::start() {
         _raft->log()->stm_manager()->add_stm(_tm_stm);
         co_return co_await _raft->start(std::move(builder));
     }
+
+    if (is_transform_offsets_topic(_raft->ntp())) {
+        vassert(topic_cfg.has_value(), "Invalid topic configuration passed.");
```

Comment:
the assert here validates if configuration has value while the message is confusing as it states that the value itself is invalid